File: 2025/Jaehyuk/2025-06-22-Asteroid-Collision.py
class Solution:
    def asteroidCollision(self, asteroids: list[int]) -> list[int]:
        stack = []
        for a in asteroids:
            while stack and a < 0 < stack[-1]:
                if -a > stack[-1]:
                    stack.pop()
                    continue
                elif -a == stack[-1]:
                    stack.pop()
                break
            else:
                stack.append(a)
        return stack

File: 2025/Heeho/2025-06-22-Asteroid-Collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<>();

        for (int asteroid : asteroids) {
            boolean alive = true;

            while (!stack.isEmpty() && asteroid < 0 && stack.peek() > 0) {
                int top = stack.peek();
                if (top < -asteroid) {
                    stack.pop();      
                } else if (top == -asteroid) {
                    stack.pop();      
                    alive = false;
                    break;
                } else {
                    alive = false;   
                    break;  
                }
            }

            if (alive) {
                stack.push(asteroid); 
            }
        }

        // 스택을 배열로 변환
        int[] result = new int[stack.size()];
        for (int i = stack.size() - 1; i >= 0; i--) {
            result[i] = stack.pop();
        }

        return result;
    }
}
